File: csharp/Chapter05/CompleteExample.cs
using System;
using System.IO;
using com.db4o;   // remember to add db4o.dll as a reference

namespace com.db4o.dg2db4o.chapter5
{
    public class CompleteExample
    {
        public static void Main(string[] args)
        {
            File.Delete("C:/complete.yap"); // reset database
            Db4o.Configure().MessageLevel(0); // 0=silent, 3 =loud
            ObjectContainer db = Db4o.OpenFile("C:/complete.yap");
            try
            {
                db.Set(new Person("Gandhi", 79));
                db.Set(new Person("Lincoln", 56));
                ObjectSet result = (ObjectSet)db.Get(new Person());
                ListResult(result); // get all

                Person p = new Person();
                p.Name = "Gandhi";
                ObjectSet result2 = (ObjectSet)db.Get(p);
                Person p2 = (Person)result2.Next();
                p2.Age = 90; // Increase Gandhis age 
                db.Set(p2);
                result2.Reset(); // reset the ObjectSet cursor before the first element
                ListResult(result2);

                db.Delete(p2); // Remove the Ghandi dataset
                ObjectSet result3 = (ObjectSet)db.Get(new Person());
                ListResult(result3); // get all

                Console.WriteLine("Press ENTER to end");
                Console.ReadLine();
            }
            finally
            {
                db.Close();
            }
        }

        private static void ListResult(ObjectSet res)
        {
            while (res.HasNext())
            {
                Console.WriteLine(res.Next());
            }
            Console.WriteLine("---------------");
        }
    }
}

File: java/src/com/db4o/dg2db4o/chapter5/CompleteExample.java

package com.db4o.dg2db4o.chapter5;

import java.io.File;
import com.db4o.Db4o;
import com.db4o.ObjectContainer;
import com.db4o.ObjectSet;

public class CompleteExample {
    public static void main(String[] args) {
        new File("C:/complete.yap").delete();
        Db4o.configure().messageLevel(0); // 0=silent, 4=loud
        ObjectContainer db = Db4o.openFile("C:/complete.yap");
        try {
            db.set(new Person("Gandhi", 79));
            db.set(new Person("Lincoln", 56));
            ObjectSet result = (ObjectSet) db.get(new Person());
            listResult(result); // get all
            
            Person p = new Person();
            p.setName("Gandhi");
            ObjectSet result2 = (ObjectSet) db.get(p);
            Person p2 = (Person) result2.next();
            p2.setAge(90); // Increase Gandhi's age 
            db.set(p2);
            result2.reset(); // reset the ObjectSet cursor before the first element
            listResult(result2);
            
            db.delete(p2); // Remove the Ghandi object
            ObjectSet result3 = (ObjectSet) db.get(new Person());
            listResult(result3); // get all
        } finally {
            db.close();
        }
    }
    
    public static void listResult(ObjectSet result) {
        while (result.hasNext()) {
            System.out.println(result.next());
        }
        System.out.println("---------------");
    }
}
